### DISCUSSION TITLE
How to clear GPU memory after training

### DISCUSSION THREAD

**Discussion by K1T00** [xamarin/TorchSharp#1367] [Q&A]
I've tried so far:

Moving model to cpu (or leave it on GPU/with and without no_grad) and trying to dispose with

```c#
model.cpu();
model.Dispose();
GC.Collect();
``` 
Also there is no
```python
torch.cuda.empty_cache()
``` 
as far as I understand. 
So how do I empty GPU memory?

**Comment by yueyinqiu** [comment]
It seems that there is no solution currently. #892

**Reply by K1T00** [reply]
Oh I see ... thank you.

**Comment by K1T00** [comment]
Searching through some of the TorchSharp projects people uploaded to github I came across projects from https://github.com/LittleLittleCloud where he implemented a call to the native (c++) function. I think pointing people to that solution/workaround would help.
